src/pages/MainPage.js

```diff
@@ -1,15 +1,64 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Outlet } from 'react-router-dom';
 import { MainFooter } from '../components/mainPageComponents/MainFooter';
-import { useAPIData } from '../hooks/useAPIData';
-import { userDataAPI } from '../api/BootcampAPI';
+import {
+  CloseButton,
+  ModalBackground,
+  ModalContent,
+  Wrapper,
+} from '../styles/styledComponents/folderStyled';
+import ModalCloseButton from '../assets/Images/ModalCloseButton.png';
+import { Modal } from '../components/modal/Modal';
 
 export default function MainPage() {
-  const { Data: userProfile } = useAPIData(userDataAPI);
+  const [modalStatus, setModalStatus] = useState(false);
+  const [modalData, setModalData] = useState({
+    ModalContent: 'AddFolderModal',
+    folderName: [
+      '전체',
+      '즐겨찾기',
+      '신규폴더',
+      '새로운폴더스',
+      'test',
+      'test2',
+      'test3',
+    ],
+    linkCounts: [3, 0, 0, 0, 0, 0, 0],
+    currentLinkCount: 3,
+    currentFolderName: '전체',
+    currentCardLink: 'https://www.google.com',
+  });
+
+  const handleModalStatus = () => {
+    setModalStatus((prev) => !prev);
+  };
+
+  const handleModalContent = (value) => {
```

Comment:
```suggestion
  const changeModalData = (value) => {
```

어떤 이벤트에 바인딩되는지를 함수명에 담기보다 어떤 행동을 하는지 함수명에 담아주세요.

코드를 읽을 때 어떤 이벤트에 바인딩되어있는지 명시되어 있으면 함수가 어떤 행동을 하는지 예측할수 없게됩니다.